src/mp2_eri_gpw.F

```diff
@@ -243,8 +245,35 @@ SUBROUTINE mp2_eri_2c_integrate_gpw(qs_env, para_env_sub, dimen_RI, mo_coeff, my
 
          CALL timestop(handle2)
 
-         ! integrate the little bastards
          offset = 0
+         ! Prepare offset ahead of OMP parallel loop
+         max_nseta = 0
+         DO iatom = 1, natom
```

Comment:
Instead of this loop you could call [get_qs_kind_set(maxnset=maxnset)](https://github.com/cp2k/cp2k/blob/master/src/qs_kind_types.F#L816).